### ISSUE TITLE
Accessing attribute of a Generic class defined by type parameters

### ISSUE THREAD

**Issue by Lotram** [microsoft/pyright#8832]
Hi, 

If I'm not mistaken, the new syntax to declare a [generic class](https://docs.python.org/3/reference/compound_stmts.html#generic-classes), is to do `MyClass[T]:` instead of `MyClass(Generic[T])`. However the following code does not give the same pyright output:

```
from typing import Generic, TypeVar


class TaskParameters:
    field: str


Params = TypeVar("Params", bound="TaskParameters")


class NewStyleTask[Params]:
    parameters: Params

    @property
    def field(self) -> str:
        return self.parameters.field


class OldStyleTask(Generic[Params]):
    parameters: Params

    @property
    def field(self) -> str:
        return self.parameters.field
```
Using pyright 1.1.377, I get:
```
  generics.py:16:32 - error: Cannot access attribute "field" for class "object*"
    Attribute "field" is unknown (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations 
```
The first class raises an error while the second does not.

**Comment by erictraut** [comment]
In your code sample, you've defined the old-style TypeVar with an upper bound, but you've omitted the upper bound for the new-style type parameter. If no upper bound is provided, the default upper bound is `object`.

If you want to make the new-style type parameter equivalent, you'd need to do this:

```python
class NewStyleTask[Params: TaskParameters]: ...
```

**Comment by Lotram** [comment]
Ok so I did miss something in the doc, thanks.
